File: frontend/src/app/_models/welcome-plugin.ts
import {DiscordRole} from './discord-role';
import {WelcomeMessage} from './welcome-message';
import {PrivateWelcomeMessage} from './private-welcome-message';
import {LeaveMessage} from './leave-message';
import {Plugin} from './plugin';

export interface WelcomePlugin extends Plugin {
  welcomeMessage: WelcomeMessage;
  privateWelcomeMessage: PrivateWelcomeMessage;
  rolesToGive: DiscordRole[];
  leaveMessage: LeaveMessage;
}

File: backend/Models/WelcomePlugin.cs
using System.Collections.Generic;

namespace backend.Models
{
    public class WelcomePlugin : Plugin
    {
        public virtual WelcomeMessage WelcomeMessage { get; set; }
        public virtual PrivateWelcomeMessage PrivateWelcomeMessage { get; set; }
        public virtual IList<DiscordRole> RolesToGive { get; set; }
        public virtual LeaveMessage LeaveMessage { get; set; }
    }
}